codalab/server/bundle_manager.py

```diff
@@ -324,13 +317,13 @@ def _schedule_run_bundles_on_workers(self, workers, user_owned):
                     continue  # Don't start this bundle yet
                 workers_list = workers.user_owned_workers(self._model.root_user_id)
 
-            workers_list = self._filter_and_sort_workers(workers_list, bundle)
+            self._deduct_worker_resources(workers_list)
```

Comment:
Felt like better abstraction to pass an up-to-date workers list into the `filter` function rather than having that function deduct its worker resources inside.